tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -402,6 +415,154 @@ shared_ptr<DuckDBPyConnection> DuckDBPyConnection::RegisterPythonObject(const st
 	return shared_from_this();
 }
 
+unique_ptr<DuckDBPyRelation> DuckDBPyConnection::ReadCSV(const string &name, const py::kwargs &kwargs) {
+	if (!connection) {
+		throw ConnectionException("Connection has already been closed");
+	}
+	auto keywords = GetKeywordArguments(kwargs);
+	BufferedCSVReaderOptions options;
+
+	// First check if the header is explicitly set
+	// when false this affects the returned types, so it needs to be known at initialization of the relation
+	if (keywords.count("header")) {
+		auto header_arg = kwargs["header"];
+
+		bool header_as_int = py::isinstance<py::int_>(header_arg);
+		bool header_as_bool = py::isinstance<py::bool_>(header_arg);
+
+		if (!header_as_int && !header_as_bool) {
+			throw InvalidInputException("read_csv only accepts 'header' as an integer, or a boolean");
+		}
+
+		if (header_as_bool) {
+			if (py::str(header_arg).is(py::str(Py_True))) {
```

Comment:
This should probably be `if (py::bool_(header_arg))` instead?

Reply:
I tried that, and I'm pretty sure that also reported true for False, hence this ugly workaround

Reply:
I used it here, and it seemed to work okay? https://github.com/duckdb/duckdb/blob/b793c7459a63570ee53d2b683276dde19667a90e/tools/pythonpkg/src/pyfilesystem.cpp#L73